pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -330,7 +333,7 @@ func (nm *NodeManager) DiscoverNode(nodeID string, searchBy cm.FindVM) error {
 				} else if !foundInternal && foundExternal {
 					klog.Warning("External address found, but internal address not found. Returning what addresses were discovered.")
 				}
-				break
+				continue
```

Comment:
Based on the comment, should we be using a `continue`? If the comment isn't valid anymore, we should update the comment to reflect the behavior.

Also if `continue` is correct, we should reset `addrs = []v1.NodeAddress{}` unless having either the external/internal is valid with IPv4/IPv6 and the having an internal/external with a IPv6/IPv4 is valid.

In others words through an example:
- external has an IPv6 address
- internal has an IPv4 address

Reply:
Regarding the comment, that doesn't reflect the behavior here, the comment needs to be changed. However, I am not sure why must break out the loop when NIC does contain a valid IPv6 and IPV4 address, in IPv6, IPv4 case. 

My understanding is if both IPv6 and IPv4 contain valid addresses, then both will be added to the Node address, which is the behavior we wanted if dual-stack is enabled.

Reply:
ahhhh I gotcha now. I thought for some reason we were supported internal/external IPv6 and a internal/external IPv4. but here we are saying literally mixed mode dual stack when you can have an external IPv6 and an internal IPv4.

we should update the comment to reflect this change and I think we are good to go